### PR TITLE
Fix ReadOnlySpanGetReferenceAndReadInteger on BigEndian #2

### PR DESCRIPTION
**Pull request by EgorBo**

### CODE REVIEW

**Comment by EgorBo** [comment]
@stephentoub @akoeplinger PTAL, I was too overconfident in my Big Endian skills in https://github.com/dotnet/runtime/pull/85969#issuecomment-1548275013

`runtime-community (Build linux-s390x Release AllSubsets_Mono)`  is green in this PR

**Review by MichalPetryka — APPROVED** [review]

File: src/libraries/System.Memory/tests/MemoryMarshal/GetReference.cs [hunk 1]
```diff
@@ -107,13 +107,13 @@ public static void ReadOnlySpanGetReferenceAndReadInteger()
...
                 0x68_00_65,
-                Unsafe.As<byte, int>(ref Unsafe.Add(ref Unsafe.As<char, byte>(
+                Unsafe.ReadUnaligned<int>(ref Unsafe.Add(ref Unsafe.As<char, byte>(
                     ref MemoryMarshal.GetReference("hello world 1".AsSpan())), 0)));
 
             Assert.Equal(BitConverter.IsLittleEndian ?
                 0x6F_00_6C_00_6C_00_65_00 :
-                0x00_65_00_6C_00_6C_00_6F,
-                Unsafe.As<byte, long>(ref Unsafe.Add(ref Unsafe.As<char, byte>(
+                0x68_00_65_00_6C_00_6C_00,
```
[the remark this answers is not in the document]
Comment (EgorBo) [reply src/libraries/System.Memory/tests/MemoryMarshal/GetReference.cs:115]: Same picture for the `int` case (that worke):
```
// "hello world 1"
// LE: 68_00_65_00_6C_00_6C_00_6F_00_20_00_77_00_6F_00_72_00_6C_00_64_00_20_00_31_00
//     00_65_00_68 (read 4 bytes)
// 
//
// BE: 00_68_00_65_00_6C_00_6C_00_6F_00_20_00_77_00_6F_00_72_00_6C_00_64_00_20_00_31
//     00_68_00_65 (read 4 bytes)
```

so chars here have swapped order ('h' is 68_00 on LE and 00_68 on BE)

[the remark this answers is not in the document]
Comment (stephentoub) [reply src/libraries/System.Memory/tests/MemoryMarshal/GetReference.cs:115]: I see. Thanks.

[the remark this answers is not in the document]
Comment (EgorBo) [reply src/libraries/System.Memory/tests/MemoryMarshal/GetReference.cs:115]: @stephentoub can you approve then? 🙂

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Memory/tests/MemoryMarshal/GetReference.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephentoub) [reply src/libraries/System.Memory/tests/MemoryMarshal/GetReference.cs:115]: GitHub wasn't letting me.  Now it is. Done.

**Pull request merged**